### PR TITLE
fix issue #56

### PR DESCRIPTION
**Pull request by coolcottontail**
https://github.com/harmony-one/go-sdk/issues/56

### CODE REVIEW

**Comment by gupadhyaya** [comment]
I think you need to handle the other fields like rate. I currently get 'Error: decimal string is empty' if I don't pass rate. @coolcottontail

**Comment by gupadhyaya** [comment]
@coolcottontail 'Error: Empty bytes' after passing --rate. So, other fields also needs to be sanity checked.

**Comment by coolcottontail** [comment]
@gupadhyaya  I fixed the initial value and use the following to test: transaction get a receipt but I am not sure whether the server will accept it.. 

./hmy staking edit-validator  --name harmony_staking  --details "this is details"  --validator-addr one1q6gkzcap0uruuu8r6sldxuu47pd4ww9w9t7tg6 --identity "staking identity" --website harmony.one --security-contact contact  -n http://s0.pga.hmny.io:9500/ --ledger --chain-id pangaea

{"transaction-receipt":"0x0ef2d95a934f67f839743d7049373e5956ef34615a71507cb2096af8b222bc8c"}

File: cmd/subcommands/staking.go [hunk 1]
```diff
@@ -314,32 +314,48 @@ Edit an existing validator"
 				return err
 			}
 
-			commisionRate, err := numeric.NewDecFromStr(commisionRateStr)
-			if err != nil {
-				return err
+			if (len(commisionRateStr) == 0) {
+				commisionRateStr = "0";
```
Comment (gupadhyaya) [inline cmd/subcommands/staking.go]: you cannot set it to zero, as the backend will through error as rate is changed more than the maxChangeRate. Any other way to skip updating this field?

Comment (gupadhyaya) [reply cmd/subcommands/staking.go]: @coolcottontail

Comment (coolcottontail) [reply cmd/subcommands/staking.go]: I can't think out any other default value for rate. This field must have a value to fill unless the edit-validator command is changed to include some flag.   so I need to mark this field as a required field, same as validator-address ? 

Also check with @rlan35  on this ?

**Comment by fxfactorial** [comment]
@coolcottontail do we still want to work on this PR?

**Comment by coolcottontail** [comment]
I can only add back the rate field as "required" field and ask the user to supply it for edit-validator,  together with all the description field.  there's no other way I can think of ...

also I found I can't compile after I pull the master...  do I need to source anything ?

pkg/keys/bls.go:120:44: undefined: shard.BlsSignature
pkg/keys/bls.go:121:20: undefined: shard.BlsSignature
pkg/keys/bls.go:132:35: undefined: shard.BlsSignature
pkg/keys/bls.go:133:10: undefined: shard.BlsSignature
pkg/keys/bls.go:178:25: undefined: "github.com/harmony-one/harmony/staking/types".BlsVerificationStr
pkg/keys/bls.go:183:19: undefined: shard.BlsSignatureSizeInBytes

**Comment by fxfactorial** [comment]
@coolcottontail ya, we fixed that build error, pull latest master, the go.mod/go.sum updated

**Comment by fxfactorial** [comment]
@coolcottontail is this still

**Comment by fxfactorial** [comment]
closed for inactivity. if functionality still needed, then @Daniel-VDM or @Cem-Harmony can implement

**Pull request closed without merging**